### ISSUE TITLE
Infernal Robotics integration with RP-1 (Rabbit Hole)

### ISSUE THREAD

**Issue by Clayell** [KSP-RO/RP-1#2464]
Ok, let me just start by saying that this is obviously a very long-term goal, and a very big rabbit hole. However, I would argue that having Infernal Robotics (IR), or some other form of functioning robotics system (not BG), working in RO and present in RP-1's tech tree would open up a lot of large opportunities for RP-1 missions.
I'll state a few missions/ideas that are only possible with a robotics system like IR, and I know there are many more:
- Spinning stages on a hinge before release (ex: Explorer 1)
- Deploying rovers from a larger lander (ex: Sojourner)
- Viable VTOL jets (ex: F-35)

Ok, saying "just integrate Infernal Robotics" is of course understating the massive amount of effort it would take to do this. Here's some things off of the top of my head that would need to be done to have IR as a well-integrated system with RP-1:
- Joint Reinforcement (one or the other)
  - Replace RO's use of KJR-Continued with KJR-Next, OR
  - Update [KJR-Continued](https://github.com/KSP-RO/Kerbal-Joint-Reinforcement-Continued) to be fully compatible with IR
- Cost (one or the other)
  - Update IR to make it compatible with RP-1's tooling system (as it has a fair amount of procedural-ness)
  - Update IR to give it costs that scale with the mass and/or size of the part and are time-accurate
- Mass
  - Update IR to have time-accurate mass for its parts, and to have the mass scale with the size of the part
- Tech Tree
  - Remove IR's science parts (surface sampler/observer) or lock them behind certain tech nodes
  - Lock improvements to mass/cost behind certain tech nodes, similar to how the tanks work

This is not an easy goal, not by any means. However, I feel that it is worthy to put this issue together to try to gather some thoughts on the idea.

**Comment by Not-a-flying-brick** [comment]
Electric props with IR parts dont work, FAR dont kount these parts "lifting"

**Comment by Clayell** [comment]
> Electric props with IR parts dont work, FAR dont kount these parts "lifting"

Even if you attach a procedural wing to them?

**Comment by Not-a-flying-brick** [comment]
yes clayell, they just produce stupid amount of torque

**Comment by Clayell** [comment]
> yes clayell, they just produce stupid amount of torque

Do you think thats an IR issue or a FAR issue though?

**Comment by Not-a-flying-brick** [comment]
FAR, with this amount of torque this should already lift off

**Comment by Not-a-flying-brick** [comment]
FAR in stock treats this issue by making dlc propeller parts "un-far", so it obeys stock rules
(which i learnt from ezsnack)

**Comment by Its-Just-Luci** [comment]
I wanna chime in and ask why not BG alongside IR? Either way, I think we'll need a proper robotics node, potentially offshot of the landing/rovers node.

**Comment by Clayell** [comment]
> why not BG alongside IR

I do not think it is worth putting time and effort to develop configs for something that people need to pay for in order to get access. (additionally, IR is better than BG in nearly every way)

**Comment by meirumeiru** [comment]
> * Joint Reinforcement (one or the other)
>   
>   * Replace RO's use of KJR-Continued with KJR-Next, OR
>   * Update [KJR-Continued](https://github.com/KSP-RO/Kerbal-Joint-Reinforcement-Continued) to be fully compatible with IR

some background about KJR

I think (but I'm not perfectly sure), that KJRcontinued is already compatible with IR (maybe I shouldn't say that and rather advertise my mod instead)... but, while I would love to see KJRnext as an accepted alternative for RO and RP-1, I don't want this because of the wrong reasons.

There are major differences between those mods today. But let me start with some history:

The mod exists mainly because KJR had no active modder during the time IRnext was first released  and KJR was incompatible with IRnext. That's why I had to do something.

But after digging into it more and more, I saw the potential to improve it and fix problems that KJR had back then and I started to do more for KJRnext and keep it as one of my main mods. After having done so much with joints for IRNext, I knew quite a bit about them and I saw how to solve the "drifting problem" that KJR had back then (I don't know how KJRcontinued handles this today).

What I also found was, that KJR versions up to that point modified the joints in a way, that they became all unrealistically strong (and at the same time unbreakable like rubber bands) and that every version simply added additional joints like crazy.

I didn't like that... because I like the challenge and every rocket bends and shakes a bit (have you ever seen the Saturn-V shake test?). But I saw, that many users of KJR wanted this. That's why I added different strength settings for KJRnext (and you can select them from the difficulty settings directly within KSP). The different options are from simply correct the unstable joints (aka "fix the problem of unity joints") up to make everything unrealistically strong (like in early/other KJR versions)... with also 2 steps between those extreme positions.

I often play with the weakest settings (which is by the way also better for performance, because it does often only add 3-20 joints in my ships while the extremest setting adds 150-200 joints) and I almost never had a problem with stability except for the problems that I want... because I want that bad design break.

In short again: Both KJR versions should be compatible, but they offer different flavors of the solution.

**Comment by meirumeiru** [comment]
> Electric props with IR parts dont work, ...

what do you mean by that?

**Comment by Not-a-flying-brick** [comment]
FAR doesn't count lift from this kind of spinning blades from my experience

meirumeiru ***@***.***> 于 2025年2月25日周二下午8:15写道：

**Comment by meirumeiru** [comment]
> FAR doesn't count lift from this kind of spinning blades from my experience

ah sorry :-D I thought you're talking about 2 different things... "electric properties" and "the missing lift of those parts" ...

**Comment by Clayell** [comment]
By the way, if anyone wants a config for having IR parts being available in RP-1 (although not placed remotely correctly), @ionite34 made a config a while ago that I've used occasionally. (https://github.com/ionite34/InfernalRoboticsNext-RO)

**Comment by Not-a-flying-brick** [comment]
Actually why dont we revive Firespitter electric props?its procedual and fits in the helicopter genre
